flytekit/extras/tasks/shell.py

```diff
@@ -97,6 +98,35 @@ def interpolate(
 T = typing.TypeVar("T")
 
 
+class MappingRevisor:
```

Comment:
Docstring please describing the example usages of this, and why it is needed for os.environ?